### DISCUSSION TITLE
2024-07-30 회의록

### DISCUSSION THREAD

**Discussion by Captain-Kim** [nbc-final-a10/TripBuddy#138] [회의록]
### 🔥  오늘의 주접

- A반 기준 Commit 압도적 1위 -> Commit 수가 서비스의 품질을 보장하진 않지만 코드 한 줄 마다 재사용성과 품질을 높이기 위한 노력은 보장할 것 같습니다. 오전 9시부터 다음 날 오전 4시까지 일 19~20 시간 가량 코딩을 하는 팀원 여러분께 감사의 말씀을 드립니다. 자축의 박수
- 여담이지만 400 커밋, 44,000줄의 코드 중 '필수 기능 구현'과 관련된 커밋은 feture/auth 외엔 없습니다. 이 부분이 걱정되긴 하지만 프로덕션이란 이런 것인가하며 깊은 깨달음을 얻는 중입니다.
- But!! 챌린지 기능 구현, 라이브러리 연결은 개인 Repo에서도 할 수 있지만 코드 재사용성을 높이기 위한 품질 개선은 이번 기회 아니면 하기 무척 어려울 것 같습니다. 이 부분에 포커스 맞추시고 많이 배워 가셨으면 좋겠습니다!

### 🔥  오늘의 일정

- **10:10** : 중간 발표 관련 발제 줌 미팅 -> 이후 국취제 기타소득 문제 등 질문 후 '총무' 선정 재논의 예정
- 모의면접 없음
- **19:00** : 2차 회의 시 디자이너님 미팅

### 🔥  1일 1PR 또는 PUSH 권장

- push만 해도 Vercel이 에러를 잡아 주니 Build Error 방지 차원에서 권장드립니다. Gloomy Sunday는 안 돼~~

### 🔥  어제 주요 변동사항 리마인드

- [검색] : 어제 야간에, 심유경 멘토님의 피드백으로, 검색 결과페이지에서 아코디언은 없어지게 되었고, 탑버튼이 추가될 것 같습니다.
- [검색] : 검색할 경우에 옵션이 다 보이고(아코디언 없어지기때문에), 스크롤이 내려가게 하여 대처
- [노티피케이션] : 페이지로 분류(여정참여 요청/수락/거절 상태, 여정찜하기, 스토리좋아요, 팔로우, 좋아요, 새메시지 등)

#### 🚀 UT 이후로 넘길것들
- 참가 모달 
- 노티피케이션
- 수락
- 강퇴
- 검색 결과 페이지
- 리치텍스트 에디터
- 인공지능 이미지 생성
- 온도지수 계산 로직
- 마이페이지/여정상세페이지/스토리상세페이지 조건부 렌더링 (블러 => (ex) 상세 정보는 가입을 하셔야 볼 수 있습니다)

### 🔥 디자이너님 미팅 가능시간

- 화 / 목 : 오후 7시이후
- 금 : 오전 10시 ~ 3시 (우리회의시간과 안겹치니까 오후에 잠시미팅)

### 🔥  개발 관련

- supabase trips, contracts 데이터 테이블 FK 설정 에러 수정 완료 -> 다른 데이터 테이블도 이상 있으면 말씀해주세요
- useSelectRegion 커스텀 훅 변동사항 2건
  - 반환값 first ~ third... 추가 => discussion 코드공유 참고
  - 커스텀 훅 호출 시 매개 변수로 `{pxHeight : 30}` 등의 값을 전달해야 함 -> 최종 목적지에 스크롤 생성하기 위함 -> but vh로 되어 있어 모바일 환경 테스트 필요함. px이 안 먹히는 트러블 슈팅 발생 -> 해결 해야 함 또는 스크롤을 뺴거나 (와이어프레임에서는 y-scroll 확인 안 됨)
- funnel 패턴 사용시 뒤로가기 앞으로가기 하면, 페이지 이동되는 문제 -> 쿼리스트링으로 해결하는 법

#### 🚀 api 담당자 리마인드

- 버디즈 : 오은
- 스토리테이블 : 오은
- 스토리라이크 : 오은
- 트립스 : 김병준
- 트립북마크 : 김병준
- 트립스관련인데검색하는라우트핸들러(필터) : 곽현정
- 메시지스 : 김민지
- 노티피케이션 : 김민지

### 🔥  **디자이너님 전달 사항**

- [검색] : 어제 야간에, 심유경 멘토님의 피드백으로, 검색 결과페이지에서 아코디언은 없어지게 되었고, 탑버튼이 추가될 것 같습니다.
- [검색] : 검색할 경우에 옵션이 다 보이고(아코디언 없어지기때문에), 스크롤이 내려가게 하여 대처
- [노티피케이션] : 페이지로 분류(여정참여 요청/수락/거절 상태, 여정찜하기, 스토리좋아요, 팔로우, 좋아요, 새메시지 등)

- **[일정] : 전체 일정 체크하고, 모바일 시안 완성여부 체크**

### 🔥 마감기한 Remind

- 필수 기능에 대해서는 최대 2024-08-01 목요일까지 완성되는 것이 좋을 듯 합니다. -> 그래야 이강민 튜터님 피드백을 받을 수 있음
- 내일 시간미정, 이강민 튜터님 개발 공수 관련 트러블 슈팅 피드백을 위해 방문하신다고 합니다.
  - 현재 진행상황에 대해서 더 정확히 파악하고 시작하는 게 좋겠습니다. 어제 전달해드린 것처럼 오늘까지 각자 Projects 수정해주세요!

### 🔥  Light House 관련

- 슬슬 LightHouse 돌려 보셔야 합니다! (시크릿 모드)
- UI든 JS든 성능 저하 되는 요소가 있다면 역으로 제안해주셔야 합니다!
  - 저대비/저시력자/대용량 자바스크립트
  - 성능 최적화는 개인의 문제가 아닌 팀의 챌린지 과제라 해결은 못 하더라도 인지하고 말씀해주시면 감사하겠습니다!

**(예시)**
![CleanShot 2024-07-30 at 19 00 55@2x](https://github.com/user-attachments/assets/da96fb45-079b-4451-9d7c-1ba31d2bf81f)

#### 공통

- 2nd, 3rd Loaction 표기 방법 통일 => 경기도 수원시
  - 데이터 컬럼 쓸 때부터

<table>
  <tr>
    <th>여정 상세 페이지</th>
    <th>채팅 알림 모달 등</th>
  </tr>
  <tr>
    <td><img src="https://github.com/user-attachments/assets/2059f8a8-ae37-480d-89f2-d5f38d9612f4" width="300" height="300"/></td>
    <td><img src="https://github.com/user-attachments/assets/6814c642-844f-41a7-8c68-30c36f804f24" width="300" height="300"/></td>
  </tr>
</table>

- 재사용 가능한 컴포넌트로 보이는 컴포넌트들, 와이어프레임과 달라지더라도 UT 까지는 통합하는 것으로

<table>
  <tr>
    <th>여정 상세 페이지</th>
    <th>알림 모달</th>
    <th>마이페이지</th>
  </tr>
  <tr>
    <td><img src="https://github.com/user-attachments/assets/4988e399-a0d1-4f55-afec-883ba9e0a519" width="300" height="300"/></td>
    <td><img src="https://github.com/user-attachments/assets/b47f37a1-a4d1-42b8-9b12-cfacab30208d" width="300" height="300"/></td>
    <td><img src="https://github.com/user-attachments/assets/be247baf-1975-4904-b379-73acf605785e" width="300" height="300"/></td>
  </tr>
</table>

#### 오은

- MascotImage 공유 컴포넌트 관련 내용

#### 김병준

- TripCard 마이페이지에서 재사용 가능하도록 모듈화 요청 (To.은님)
  - '참여하기' -> '상세보기' 라벨 변경 가능하도록.
  - onClick 속성 추가

### 🔥  내일 예정 일정

- 아침 회의때, UT 비용 관련 논의해야 합니다. (담당자 오은)
- 시간 미정, 이강민 튜터님 피드백 예정

**Comment by Captain-Kim** [comment]
## 김병준

### 게시글 작성 기능 구현은 완료

- 라우트 작성까지 완료했으나.. 아래가 남음
  - 펜딩, 에러 처리 => 완료
  - 유효성 검사
  - WebP 이미지 처리 유틸 함수 제작
  - 추가 옵션 선택용 커스텀 훅 제작 (성별, 나이) => 이미 제작하신 것이 있지만 제작하는 게 더 빠를 것 같아 별도 제작 후 리팩토링

### useFunnelButton 커스텀 훅 관련

- useNextButton 커스텀 훅 onClick 속성 추가되었다고 들은 것 같은데... 보이지 않아아서 옵셔널 Props로 추가함 (게시글 작성 버튼 이벤트 핸들러 연결 목적) => 전달 완료, 본인이 부여한 onClick 속성으로 통일하기로 (To. 은님)

<img src="https://github.com/user-attachments/assets/0c5a86f2-0617-4d94-af6d-2f59483df1d9" width="300" height="300"/>

- Funnel Pattern 조건부 렌더링을 위한 커스텀 훅 수정 불가피 해 보임
  - 이 문제 처리해야 함. 퍼널 패턴에서 글 작성 페이지만 버튼 텍스트가 다름. 최초 설계에서 빗나가는 골치 아픈 트러블슈팅 발생 => 퍼널 패턴 수정 불가피할 듯 <<충돌주의>>

<img src="https://github.com/user-attachments/assets/b165149d-d68e-478e-bef4-90104f322276" width="300" height="300"/>

- 여행지 검색 컴포넌트에서 유저가 최종 목적지를 선택한 상태에서 다른 아이템을 선택하기 전까지 state가 유지됨 -> 이 상태에서 다음 버튼을 누르면 유저는 어떤 것을 클릭했는지 모르는 상태로 다음 시퀀스로 넘어감 -> 텍스트로 state 상태를 알림을 띄우거나 1, 2단계 Location 선택 시 null로 초기화 하는 로직을 구현하거나
- 이 부분 hover or clicked로 스타일링을 추가하고자 했으나 개발 서버에서 보였다가 안 보였다가 테스트 환경이 이상함 -> 일단은 두고 시간 여유 되시는 분께 help 요청 (이미 hover 속성은 먹히는 것 같은 위치에 추가해 놓음)

### 마이페이지 버디즈 테마 관련

- 기존 Chips 재사용이 불가능해보임 -> 수정 가능하나 공수가 꽤 예상되는 바 은님과 긴밀히 협의할 예정

**Comment by eunohhh** [comment]
## 오은
- 온보딩은 isOnboarding이 false일 때 한번만 되는 것인데 마지막 step 에서 한번에 fetch 하게 되면 
- 모종의 이유로 에러가 발생해서 시퀀스가 중단 되었을 경우에 아무 정보도 fetch 하지 못한 상태가 되기 때문에 
- 각 step 마다 fetch 하는 것으로 결정
